flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -406,6 +407,43 @@ public void onError(Throwable error) {
 		onFatalError(error);
 	}
 
+	@Override
+	public void onContainerStarted(ContainerId containerId, Map<String, ByteBuffer> allServiceResponse) {
+		log.info("The container {} started.", containerId);
+	}
+
+	@Override
+	public void onContainerStatusReceived(ContainerId containerId, ContainerStatus containerStatus) {
+	}
+
+	@Override
+	public void onContainerStopped(ContainerId containerId) {
+		log.info("The container {} is stopped.", containerId);
+	}
+
+	@Override
+	public void onStartContainerError(ContainerId containerId, Throwable t) {
+		log.error("Could not start TaskManager in container {}.", containerId, t);
+
+		// release the failed container
+		YarnWorkerNode yarnWorkerNode = workerNodeMap.remove(new ResourceID(containerId.toString()));
+		resourceManagerClient.releaseAssignedContainer(containerId);
+		// ask for a new one
+		requestYarnContainer(yarnWorkerNode.getContainer().getResource(), yarnWorkerNode.getContainer().getPriority());
+	}
+
+	@Override
+	public void onGetContainerStatusError(ContainerId containerId, Throwable t) {
+		log.error("Error occurred during get the container {} status.", containerId, t);
+		onFatalError(t);
```

Comment:
I think maybe this is not a fatal error.

Reply:
change